.github/workflows/go.yml

```diff
@@ -374,7 +376,7 @@ jobs:
 
   linux-arm:
     name: ARM64 Debian 11 GO ${{ matrix.go }}
-    runs-on: ["arm", "linux"]
+    runs-on: ["self-hosted", "linux"]
```

Comment:
@raulcd Should we prefer `self-hosted` to `arm` here?

Reply:
I was getting a syntax error in my IDE for having "arm" there based on GHA json schema. That's why I changed it.

Reply:
I see.
I'll wait for @raulcd.

Reply:
I don't mind using `self-hosted` but I think we should also use the architecture in case we want to add other self-hosted architectures in the future.

Reply:
In the interests of getting the Go tests running again, I'm going to merge this but if @raulcd or @assignUser thinks I should change it back to "arm", just comment here and I'll do so.

Reply:
Yeah what @raulcd said as it also matches the official recommendation.

Reply:
so, it should be `["self-hosted", "linux", "arm"]` ?